dask_expr/_shuffle.py

```diff
@@ -1042,6 +1044,10 @@ def _simplify_up(self, parent, dependents):
                     return NLargest(self.frame, n=parent.n, _columns=self.by)
                 else:
                     return NLargestSlow(self.frame, n=parent.n, _columns=self.by)
+            else:
```

Comment:
I think you can just remove Lines 1037-1046 or does this fail tests? Same for NLast Also ok as a follow up